File: PaymentSDK/UIHostObjCApp/sdk/PaymentSDKDelegate.m
//
//  PaymentSDKDelegate.m
//  UIHostObjCApp
//
//  Created by Rahul Dhuri on 30/05/19.
//  Copyright © 2019 Network International. All rights reserved.
//

#import "PaymentSDKDelegate.h"
#import "OrderService.h"
#import "Amount.h"
#import "CartData.h"

@implementation PaymentSDKDelegate

- (void)authorizationCompletedWithStatus:(enum AuthorizationStatus)status {
    NSLog(@"Auth Completed with status %ld", (long)status);
}

- (void)authorizationStarted {
    
}

- (void)beginAuthorizationWithDidSelect:(PaymentMethod * _Nonnull)paymentMethod handler:(void (^ _Nonnull)(PaymentAuthorizationLink * _Nullable))completion {
    NSLog(@"Create order");
    
    Amount *amount = [Amount initWithCurrency: [[CartData sharedInstance] currency] andValue: [[CartData sharedInstance] cartTotal]];
    [OrderService create: amount withAction: @"AUTH" withCompletion:^(OrderResponse * _Nonnull order, NSError * _Nonnull error) {
        if (error == nil) {
            if (order) {
                PaymentAuthorizationLink *authLink = [[PaymentAuthorizationLink alloc]initWithHref:order.paymentAuthorizationUrl code:order.code];
                completion(authLink);
            }
        } else {
            NSLog(@"Failed to get authorization link with error : %@", error);
            completion(nil);
        }
    }];
}

- (void)paymentCompletedWith:(enum PaymentStatus)status {
    
}

- (void)paymentStarted {
    
}

@end

File: PaymentSDK/UIHostApp/sdk/PaymentSDKDelegate.swift
//
//  PaymentSDKDelegate.swift
//  UIHostApp
//
//  Created by Niraj Chauhan on 5/5/19.
//  Copyright © 2019 Network International. All rights reserved.
//

import Foundation
import PaymentSDK

final class PaymentSDKDelegate : PaymentDelegate {
    func beginAuthorization(didSelect paymentMethod: PaymentMethod, handler completion: @escaping (PaymentAuthorizationLink?) -> Void) {
        print("Create order")
        let amount = Amount(currencyCode: ViewController.currency, value: ViewController.cartTotal)
        OrderService.create(amount: amount, action: "AUTH"){
            (orderCreateResponse, e) in
            if let error = e {
                print(error)
                return
            }
            if let order = orderCreateResponse {
                let authLink = PaymentAuthorizationLink(href: order.paymentAuthorizationUrl, code: order.code)
                completion(authLink)
            }
            
        }
    }
    
    func authorizationStarted() {
        print("Auth started")
    }
    
    func authorizationCompleted(withStatus status: AuthorizationStatus) {
        print("Auth Completed with status \(status)")
    }
    
    func paymentStarted() {
        print("Payment started")
    }
    
    func paymentCompleted(with status: PaymentStatus) {
        print("Payment completed with status: \(status)")
    }
}
